pkg/operator2/configobservation/routersecret/observe_router_secret.go

```diff
@@ -0,0 +1,86 @@
+package routersecret
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-authentication-operator/pkg/operator2/configobservation"
+)
+
+func ObserveRouterSecret(genericlisters configobserver.Listers, recorder events.Recorder, previouslyObservedConfig map[string]interface{}) (map[string]interface{}, []error) {
```

Comment:
Maybe consider refactoring this to separate unmarshalling from certificate retrieval from marshalling, and add testing for each step? Working directly with unstructured is always a bit dangerous, and the lack of testing here means the only way to catch problems will be by deploying to a real cluster (read: costly).

I followed this approach when adding an observer for `ServiceAccountIssuer` in support of bound token configuration:

https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/configobservation/auth/auth_serviceaccountissuer.go
https://github.com/openshift/cluster-kube-apiserver-operator/blob/master/pkg/operator/configobservation/auth/auth_serviceaccountissuer_test.go